### PR TITLE
[MM-28664] model/integration_action: fix panic for uncomparable types

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
In such cases we end up comparing two `map[string]interface{}`s. When we try to compare, it causes a panic: `runtime error: comparing incomparable type map[string]interface {}`. We can use `reflect.DeepEqual()` instead.

PS: `reflect.Equal()` would be pretty slow (10 times according to my benchmarks for a single comparison) but it is the safest one. We can review this approach again if it has a significant performance impact.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28664

### CODE REVIEW

**Comment by agnivade** [comment]
This code path is only taken for slack webhooks right? If so, then it should be fine.

**Review by agnivade — APPROVED** [review]

File: model/integration_action_test.go [hunk 1]
```diff
@@ -161,3 +161,27 @@ func TestSubmitDialogResponseToJson(t *testing.T) {
 		assert.Nil(t, r)
 	})
 }
+
+func TestPostActionIntegrationEquals(t *testing.T) {
```
Comment (streamer45) [inline model/integration_action_test.go:165]: I'd suggest to also add a comparable type case and one where the contents do not match.

File: model/integration_action_test.go [hunk 2]
```diff
@@ -161,3 +161,27 @@ func TestSubmitDialogResponseToJson(t *testing.T) {
...
+			Context: map[string]interface{}{
+				"a": map[string]interface{}{
+					"a": 0,
+				},
+			},
+		},
+	}
+	if !pa1.Equals(pa2) {
+		t.Error("should be equal")
+	}
```
Comment (streamer45) [inline model/integration_action_test.go:186]: Any reason for not using`require.True()` here?

**Review by streamer45 — COMMENTED** [review]
Looks good! :+1: 
Left one suggestion to improve the tests.

File: model/integration_action_test.go (hunk 2 above)
Comment (isacikgoz) [reply model/integration_action_test.go]: Good point 👍

**Review by streamer45 — APPROVED** [review]
Very good :+1:

**Comment by isacikgoz** [comment]
@lindalumitchell This yet another sentry report that caused by some unusual payload. I'd suggest to rely on automated tests on this.

**Review by lindalumitchell — APPROVED** [review]

**Pull request merged**